### PR TITLE
Write slices

### PR DESCRIPTION
**Pull request by jamiesandenr**
Once again, sorry about all the formatting from black.  At least it should only happen this once.

### CODE REVIEW

**Comment by jamiesandenr** [comment]
Do check, but I think the only actual code changes are the new methods:
`write_stack`
`_write_group_stack`
`_get_stack_length`
`_get_slice_from_stack`
and the version number.

**Comment by mhantke** [comment]
Would it make sense to make a separate PR that only shows the changes by black and then merge after that goes in?

**Comment by jamiesandenr** [comment]
> Would it make sense to make a separate PR that only shows the changes by black and then merge after that goes in?

Will do.

**Comment by jamiesandenr** [comment]
Black stuff all merged in now, so this is ready for review.

File: setup.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 setup(
     name="h5writer",
-    version="0.6.0",
+    version="0.7.0",
     description="Tool for writing HDF5 files",
     author="Max F. Hantke, Benedikt Daurer, Filipe R. N. C. Maia",
```
Comment (mhantke) [inline setup.py]: You should add your name :)

File: h5writer/h5writer_single.py [hunk 2]
```diff
@@ -45,6 +61,25 @@ def write_slice(self, data_dict):
         # Update of maximum index
         self._i_max = max([self._i, self._i_max])
 
+    def write_stack(self, stack: dict) -> None:
```
Comment (mhantke) [inline h5writer/h5writer_single.py]: We already use the term `stack` for the entire stack. How about naming this method `write_concatenated_slices` or `write_substack` to be more clear?

File: h5writer/h5writer.py [hunk 3]
```diff
@@ -103,6 +124,41 @@ def _write_group(self, D, group_prefix="/"):
                 else:
                     self._f[name][self._i, :] = data[:]
 
+    def _write_group_stack(self, stack: dict, length: int, group_prefix="/") -> None:
```
Comment (mhantke) [inline h5writer/h5writer.py]: See comment below

File: h5writer/h5writer.py [hunk 4]
```diff
@@ -42,6 +44,25 @@ def __init__(self, filename, chunksize, compression):
             self._create_dataset_kwargs["compression"] = compression
         self._initialised = False
 
+    @staticmethod
+    def _get_stack_length(stack: dict) -> int:
```
Comment (mhantke) [inline h5writer/h5writer.py]: See comment below

**Review by mhantke — APPROVED** [review]

**Pull request merged**